### PR TITLE
[Fix] Possible null reference.

### PR DESCRIPTION
**Pull request by BaronGreenback**
Fixes a comparison with null causing a NullReference exception.

### CODE REVIEW

File: MediaBrowser.Common/Plugins/LocalPlugin.cs [hunk 1]
```diff
@@ -106,6 +106,11 @@ public override int GetHashCode()
         /// <inheritdoc />
         public bool Equals(LocalPlugin other)
         {
+            if (!(other is LocalPlugin))
```
Comment (cvium) [inline MediaBrowser.Common/Plugins/LocalPlugin.cs:109]: shouldn't it just be `other == null`? It's either null or LocalPlugin (or a subclass of LocalPlugin)

Comment (BaronGreenback) [reply MediaBrowser.Common/Plugins/LocalPlugin.cs:109]: That's what i did first time around. But @crobibero overrode == which calls Equals, so you end up in an endless loop.

Comment (cvium) [reply MediaBrowser.Common/Plugins/LocalPlugin.cs:109]: Use https://docs.microsoft.com/en-us/dotnet/api/system.object.referenceequals?view=net-5.0

File: MediaBrowser.Common/Plugins/LocalPlugin.cs [hunk 2]
```diff
@@ -106,6 +106,11 @@ public override int GetHashCode()
         /// <inheritdoc />
         public bool Equals(LocalPlugin other)
         {
+            if (object.ReferenceEquals(other, null))
```
Comment (cvium) [inline MediaBrowser.Common/Plugins/LocalPlugin.cs:109]: Maybe add a comment here about why it's necessary

**Review by cvium — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**